tests/integration/test_enrollment/conftest.py

```diff
@@ -0,0 +1,333 @@
+                                 connection_protocol='SSL', func=receiver_callback)
+
+
+@pytest.fixture(scope="module")
+def create_certificates():
+    cert_controller = CertificateController()
+    cert_controller.get_root_ca_cert().sign(cert_controller.get_root_key(), cert_controller.digest)
+    cert_controller.store_private_key(cert_controller.get_root_key(), AGENT_KEY_PATH)
+    cert_controller.store_ca_certificate(cert_controller.get_root_ca_cert(), AGENT_CERT_PATH)
+
+
+@pytest.fixture(scope="function")
+def configure_socket_listener():
+    """Configures the socket listener to start listening on the socket."""
+    socket_listener.start()
+    socket_listener.listener.set_ssl_configuration(connection_protocol=ssl.PROTOCOL_TLSv1_2,
+                                                   certificate=AGENT_CERT_PATH,
+                                                   keyfile=AGENT_KEY_PATH,
+                                                   options=None,
+                                                   cert_reqs=ssl.CERT_OPTIONAL)
+
+    while not socket_listener.queue.empty():
+        socket_listener.queue.get_nowait()
+    socket_listener.event.clear()
+
+    yield
+    socket_listener.shutdown()
+
+
+# Wazuh conf
+
+def get_temp_yaml(param):
+    """Creates a temporal config file."""
+    temp = os.path.join(test_data_path, 'temp.yaml')
+    with open(configurations_path, 'r') as conf_file:
+        enroll_conf = {'enrollment': {'elements': []}}
+        for elem in param:
+            if elem == 'password':
+                continue
+            enroll_conf['enrollment']['elements'].append({elem: {'value': param[elem]}})
+        print(enroll_conf)
+        temp_conf_file = yaml.safe_load(conf_file)
+        temp_conf_file[0]['sections'][0]['elements'].append(enroll_conf)
+    with open(temp, 'w') as temp_file:
+        yaml.safe_dump(temp_conf_file, temp_file)
+    return temp
+
+
+def override_wazuh_conf(configuration, test):
+    """Re-writes Wazuh configuration file with new configurations from the test case.
+    Args:
+        configuration (dict): Dictionary with the configuration to overwrite.
+        test (str): Name of the current test.
+    """
+    parse_configuration_string(configuration)
+    # Configuration for testing
+    temp = get_temp_yaml(configuration)
+    conf = load_wazuh_configurations(temp, test, )
+    os.remove(temp)
+
+    test_config = set_section_wazuh_conf(conf[0]['sections'])
+    # Set new configuration
+    write_wazuh_conf(test_config)
+
+
+# Keys file
+
+@pytest.mark.parametrize('test_case', [case for case in tests])
+@pytest.fixture(scope="function")
+def set_keys(test_case):
+    """Writes the keys file with the content defined in the configuration.
+    Args:
+        test_case (dict): Current test case.
+    """
+    keys = test_case.get('pre_existent_keys', [])
+    if not keys:
+        return
```

Comment:
Why `if not` instead of:

```
if keys:
  with open(CLIENT_KEYS_PATH, "w") as keys_file:
            for key in keys:
                keys_file.writelines(key)
```

What are we returning?

Reply:
Fixed